Userland/Libraries/LibWeb/Painting/GradientPainting.cpp

```diff
@@ -257,7 +257,7 @@ void paint_linear_gradient(PaintContext& context, DevicePixelRect const& gradien
     auto gradient_length_px = calulate_gradient_length(gradient_rect.size(), sin_angle, cos_angle);
     DevicePixelPoint offset { cos_angle * (gradient_length_px / 2), sin_angle * (gradient_length_px / 2) };
     auto center = gradient_rect.translated(-gradient_rect.location()).center();
-    auto start_point = center.to_type<float>() - offset.to_type<float>();
+    auto start_point = center.to_type<int>().to_type<float>() - offset.to_type<int>().to_type<float>();
```

Comment:
What's with the `.to_type<int>().to_type<float>()`? I've seen this in a few places, seems a little confusing.

Reply:
(Maybe) we could add our `cast` function to LibWeb that could handle the special cases like this. So this could be just something like `cast_to<float>(center)`

Reply:
I've gone with adding `IntPoint` overloads to the Painter methods that are used. See: https://github.com/SerenityOS/serenity/pull/16366